packages/swi_prolog.rb

```diff
@@ -0,0 +1,58 @@
+require 'package'
+require 'fileutils'
+
+class Swi_prolog < Package
+  description 'SWI-Prolog offers a comprehensive free Prolog environment. Since its start in 1987, SWI-Prolog development has been driven by the needs of real world applications. SWI-Prolog is widely used in research and education as well as commercial applications. Join over a million users who have downloaded SWI-Prolog.'
+  homepage 'https://www.swi-prolog.org/'
+  version '8.2.4'
+  compatibility 'all'
+  source_url 'https://www.swi-prolog.org/download/stable/src/swipl-8.2.4.tar.gz'
+  source_sha256 'f4bcc78437f9080ab089762e9e6afa7071df7f584c14999b92b9a90a4efbd7d8'
+
+  binary_url ({
+  })
+  binary_sha256 ({
+  })
+
+  depends_on 'zlibpkg'
+  depends_on 'openssl'
+  depends_on 'gmp'
+  depends_on 'readline'
+  depends_on 'libarchive'
+  depends_on 'ncurses'
+  depends_on 'libdb'
+  depends_on 'acl'
+  depends_on 'ninja' => :build
+  depends_on 'cmake' => :build
+  depends_on 'llvm' => :build
```

Comment:
No need to include most of these dependencies since they're all core packages.  Only need:
```
depends_on 'libarchive'
depends_on 'libdb'
```

Reply:
I removed all dependencies except libarchive, libdb, and libyaml (which I added a minute ago when I realized we had it available). Please let me know if that dependency set looks good.

Reply:
Dirty script to check for dependencies in your package file:
```
#!/bin/bash
curl -Lf https://github.com/skycocker/chromebrew/raw/master/tools/core_packages.txt > /tmp/core_packages
for i in $(grep depends_on $1 | awk '{print $2}' | sed "s/'//g"); do grep $i /tmp/core_packages ; done
```